### PR TITLE
Implement `plot_intermediate_values` and `_get_intermediate_plot` with Matplotlib backend.

### PR DESCRIPTION
**Pull request by ytknzw**
This PR proposes to implement a ``matplotlib`` backend for plotting intermediate values based on PR #1756 .

## Motivation
To provide another visualization option for users, who have only ``plotly`` now.
See issue #1539 and PR #1707 for the original motivation.

## Description of the changes
Create  ``optuna.visualization.matplotlib._intermediate_values.plot_intermediate_values`` and ``optuna.visualization.matplotlib._intermediate_values._get_intermediate_plot``.

### CODE REVIEW

File: optuna/visualization/matplotlib/_intermediate_values.py [hunk 1]
```diff
@@ -1,11 +1,72 @@
...
+_logger = get_logger(__name__)
 
 
 @experimental("2.2.0")
-def plot_intermediate_values() -> Axes:
-    raise NotImplementedError("To be implemented soon.")
+def plot_intermediate_values(study: Study) -> "Axes":
+    """Plot intermediate values of all trials in a study with Matplotlib.
+
+    .. seealso::  optuna.visualization.plot_intermediate_values
```
Comment (HideakiImamura) [inline optuna/visualization/matplotlib/_intermediate_values.py:19]: This `.. seealso::` section can be made more descriptive.
```suggestion
    .. seealso::  
        Please refer to :func:`~optuna.visualization.plot_intermediate_values` for an example.
```

**Review by HideakiImamura — COMMENTED** [review]
Sorry for the late review. The change almost LGTM. I have a minor comment on the doc.

**Comment by ytknzw** [comment]
Thank you for your review! I've updated the doc.

File: optuna/visualization/matplotlib/_intermediate_values.py [hunk 2]
```diff
@@ -1,11 +1,73 @@
...
 
 
 @experimental("2.2.0")
-def plot_intermediate_values() -> Axes:
-    raise NotImplementedError("To be implemented soon.")
+def plot_intermediate_values(study: Study) -> "Axes":
+    """Plot intermediate values of all trials in a study with Matplotlib.
+
+    .. seealso::
+        Please refer to :func:`~optuna.visualization.plot_intermediate_values` for an example.
```
Comment (HideakiImamura) [inline optuna/visualization/matplotlib/_intermediate_values.py:20]: Thanks for the swift action! According to the https://github.com/optuna/optuna/pull/1899#discussion_r499364752, we should show the module path. Could you update it?
```suggestion
        Please refer to :func:`optuna.visualization.plot_intermediate_values` for an example.
```

Comment (ytknzw) [reply optuna/visualization/matplotlib/_intermediate_values.py]: Thanks!

**Review by HideakiImamura — APPROVED** [review]
LGTM!

**Comment by hvy** [comment]
Thanks a lot for the PR! The code changes LGTM. I'm wondering if we shouldn't upda the .rst https://github.com/optuna/optuna/blob/master/docs/source/reference/visualization/matplotlib.rst as well. (C.f. https://github.com/optuna/optuna/pull/1847) It could be done in a follow up but it's just a matter of a single line.

**Comment by ytknzw** [comment]
Thank you for reviewing! I've added the line to matplotlib.rst.

**Comment by hvy** [comment]
It should be easier for user to find it now, thanks. LGTM!

**Review by hvy — APPROVED** [review]

**Pull request merged**